collective/documentviewer/convert_all.py

```diff
@@ -0,0 +1,64 @@
+from collective.documentviewer.async import asyncInstalled
+from collective.documentviewer.async import celeryInstalled
+from collective.documentviewer.async import queueJob
+from collective.documentviewer.settings import GlobalSettings
+from collective.documentviewer.settings import Settings
+from collective.documentviewer.utils import allowedDocumentType
+from logging import getLogger
+from Products.CMFCore.utils import getToolByName
+from zope.annotation.interfaces import IAnnotations
+from zope.component.hooks import getSite
+from zope.globalrequest import getRequest
+
+import transaction
+
+
+logger = getLogger('collective.documentviewer')
+
+
+def convert_all(only_unconverted=True):
+    """Convert all files.
+    Defaults to convert only files, which haven't been converted yet.
+    """
+    site = getSite()
+
+    qi = getToolByName(site, 'portal_quickinstaller', None)
+    if not qi:
+        return
+    if not qi.isProductInstalled('collective.documentviewer'):
+        return
+    if getRequest().get('plone.app.contenttypes_migration_running', False):
```

Comment:
This will fail if getRequest() returns None (same issue fixed in #72). I suggest to use `getRequest() or {}`.
Also what about allowing the user to optionally pass an obj?
I would change the signature of the function to `def convert_all(obj=None, only_unconverted=True):` and later do `if obj == None: obj = getSite()`.
In this way we can run the conversion just on a particular folder, which might be a really nice idea!